sklearn/linear_model/stochastic_gradient.py

```diff
@@ -66,6 +67,29 @@ def __init__(self, loss, penalty='l2', alpha=0.0001, C=1.0,
         self.warm_start = warm_start
         self.average = average
 
+        if n_iter is not None:
+            warnings.warn("n_iter parameter is deprecated in 0.19 and will be"
+                          " removed in 0.21. Use max_iter and tol instead.",
+                          DeprecationWarning)
+            # Same behavior as before 0.19
+            self.max_iter = n_iter
+            self.tol = -np.inf
+
+        elif tol is None and max_iter is None:
+            warnings.warn(
+                "max_iter and tol parameters have been added in 0.19. If both "
+                "are left unset, they default to max_iter=5 and tol=-inf. "
+                "If you set one of them, the other one is set to its future "
+                "default value: From 0.21, default max_iter will be 100, "
+                "and default tol will be 1e-3.", FutureWarning)
```

Comment:
I would put `type(self)` somewhere in the error message. The user is likely to get a warning when running existing code after an upgrade without necessarily remembering that `SGDClassifier / Regressor` is used in his/her code.